os/drivers/lcd/mipi_lcd.c

```diff
@@ -399,6 +405,38 @@ static int lcd_setcontrast(FAR struct lcd_dev_s *dev, unsigned int contrast)
 	return OK;
 }
 
+FAR void lcd_set_logo(FAR struct lcd_dev_s *dev)
+{
+	int logo_arr_index = 0;
+	int lcd_data_index = CONFIG_LCD_XRES * (CONFIG_LCD_YRES - CONFIG_LCD_LOGO_YRES) + (CONFIG_LCD_XRES - CONFIG_LCD_LOGO_XRES);
+	int lcd_data_col_count = 0;
+	FAR struct mipi_lcd_dev_s *priv = (FAR struct mipi_lcd_dev_s *)dev;
+
+#if !defined(CONFIG_LCD_SW_ROTATION)
+	lcd_data_logo = (uint8_t *)kmm_malloc(CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2 + 1);
+	if (!lcd_data_logo) {
+		lcddbg("ERROR: LCD logo data memory allocation failed\n");
+		return;
+	}
+#endif
+	for (int i = 0; i < CONFIG_LCD_XRES * CONFIG_LCD_YRES * 2; i++) {
+		lcd_data_logo[i] = 0;
+	}
```

Comment:
Is this necessary?
The while loop (line 426) does not fill (set) data in every index?

Reply:
we are first filling the buffer with all 0's which is black color.
then add logo in specific index in buffer.

Reply:
Oh. ok. let's leave that description.

Reply:
I have added the description as comment in code.